### DISCUSSION TITLE
I can't deploy React app to github pages

### DISCUSSION THREAD

**Discussion by ZsuzsaMano** [community/community#21764] [Other Feature Feedback, Questions, & Ideas]
<p>I’m trying since 3 days to deploy my react app on Github pages.</p>
<p>I followed the steps of this:</p>
<p><a href="https://dev.to/yuribenjamin/how-to-deploy-react-app-in-github-pages-2a1f" rel="nofollow noopener">https://dev.to/yuribenjamin/how-to-deploy-react-app-in-github-pages-2a1f</a></p>
<p>it builds, but woudnt push to github.</p>
<p>resume@0.1.0 deploy /home/zsuzsa/Asztal/Projects/resume<br>
&gt; gh-pages -d build</p>
<p>Cloning into ‘node_modules/gh-pages/.cache/git@github.com!ZsuzsaMano!<a href="http://ZsuzsaMano.github.io" rel="nofollow noopener">ZsuzsaMano.github.io</a>’…<br>
Permission denied (publickey).<br>
fatal: Could not read from remote repository.</p>
<p>Please make sure you have the correct access rights<br>
and the repository exists.</p>
<p>npm ERR! code ELIFECYCLE<br>
npm ERR! errno 1</p>
<p>npm ERR! resume@0.1.0 deploy: <code>gh-pages -d build</code><br>
npm ERR! Exit status 1<br>
npm ERR!<br>
npm ERR! Failed at the resume@0.1.0 deploy script.<br>
npm ERR! This is probably not a problem with npm. There is likely additional logging output abov<br>
e.</p>
<p>npm ERR! A complete log of this run can be found in:<br>
npm ERR! /home/zsuzsa/.npm/_logs/2019-11-17T18_13_43_540Z-debug.log</p>
<p>then after a lot of different trys I tried to just drag the build folder to my repo, after the 3rd time, it finally worked.</p>
<p>I set githubpages in settings, but on</p>
<p><a href="https://zsuzsamano.github.io/Resume/" rel="nofollow noopener">https://zsuzsamano.github.io/Resume/</a>          I still only get Error  404</p>
<p><a href="https://github.com/ZsuzsaMano/Resume" rel="nofollow noopener">https://github.com/ZsuzsaMano/Resume</a> is my repo.</p>
<p>Im already quite desperate… greatful for any ideas</p>
<p>Im using Ubuntu 16.04 and Atom</p>

**Comment by ZsuzsaMano** [comment, marked as answer]
<p>I change ssh to https in package.json homepage</p>
<p>That made the magic.</p>

**Reply by navom1** [reply]
how do I do this?

**Comment by kcbarter** [comment]
<p>Having the same issue but I have https for the homepage already</p>
<blockquote>
<p><a class="mention" href="/zsuzsamano">@zsuzsamano</a> wrote:</p>
</blockquote>
<blockquote>
<p>I change ssh to https in package.json homepage</p>
</blockquote>
<blockquote>
<p>That made the magic.</p>
</blockquote>
<blockquote>
<p> </p>
</blockquote>

**Comment by SerMedvid** [comment]
<p>I solved the issue by changing SSH connection to HTTPS,</p>
<p>remove SSH origin and add HTTPS, worked for me</p>

**Comment by pak11273** [comment]
<p>Unacceptable.  I see plenty of developers not having to switch over to https and getting this to work.  I don’t want to have to keep putting my username and password in every time I’m deploying.  </p>
<p>I just solved this problem…</p>
<p>If you use a passphrase for git authentication then this solution might work for you.  I wrote this answer on StackExchange.  Check it out here: <a href="https://superuser.com/questions/1435950/git-gh-pages-deployment-permissions-error-permission-denied-publickey-fatal/1539414#1539414" rel="nofollow noopener">https://superuser.com/questions/1435950/git-gh-pages-deployment-permissions-error-permission-denied-publickey-fatal/1539414#1539414</a></p>
